elasticgraph-schema_definition/lib/elastic_graph/schema_definition/results.rb

```diff
@@ -242,7 +242,9 @@ def identify_extra_update_targets_by_object_type_name
         sourced_field_errors = [] # : ::Array[::String]
         relationship_errors = [] # : ::Array[::String]
 
-        state.object_types_by_name.except("Query").values.each_with_object(::Hash.new { |h, k| h[k] = [] }) do |object_type, accum|
+        state.object_types_by_name.except("Query").values.each_with_object(
+          ::Hash.new { |h, k| h[k] = [] } #: Hash[untyped, Array[SchemaArtifacts::RuntimeMetadata::UpdateTarget]]
```

Comment:
This type annotation for the `Hash.new` is really annoying, but couldn't find a better way to type check it.
Assigning the accumulator to another local variable with type annotations, and destructively update them would make more sense. But I tried to keep the original code structure here.

Reply:
I don't think we need to keep using `each_with_object(Hash.new { ... })` if it's hard to type check.

Reply:
Rewrote some trivial `each_with_object` calls with `each`. One in `#identify_extra_update_targets_by_object_type_name` is left because it's complicated.